src/main/java/org/jabref/logic/citationkeypattern/BracketedPattern.java

```diff
@@ -846,51 +846,36 @@ static String allAuthors(AuthorList authorList) {
+                authorList.getAuthor(numberOfAuthors - 1).equals(Author.OTHERS);
+
+        if (numberOfAuthors == 1 || andOthersPresent) {
+            // Single author or "and others" case
+            String lastName = getLastName(authorList.getAuthor(0));
+            alphaStyle.append(lastName, 0, Math.min(2, lastName.length()));
         } else {
-            maxAuthors = MAX_ALPHA_AUTHORS - 1;
-            maxAuthorsExceeded = true;
-        }
-
-        if (authorList.getNumberOfAuthors() == 1) {
-            String[] firstAuthor = authorList.getAuthor(0).getNamePrefixAndFamilyName()
-                                             .replaceAll("\\s+", " ").trim().split(" ");
-            // take first letter of any "prefixes" (e.g. van der Aalst -> vd)
-            for (int j = 0; j < (firstAuthor.length - 1); j++) {
-                alphaStyle.append(firstAuthor[j], 0, 1);
-            }
-            // append last part of last name completely
-            alphaStyle.append(firstAuthor[firstAuthor.length - 1], 0,
-                    Math.min(3, firstAuthor[firstAuthor.length - 1].length()));
-        } else {
-            boolean andOthersPresent = authorList.getAuthor(maxAuthors - 1).equals(Author.OTHERS);
-            if (andOthersPresent) {
-                maxAuthors--;
-            }
-            List<String> vonAndLastNames = authorList.getAuthors().stream()
-                                                     .limit(maxAuthors)
-                                                     .map(Author::getNamePrefixAndFamilyName)
-                                                     .toList();
-            for (String vonAndLast : vonAndLastNames) {
-                // replace all whitespaces by " "
-                // split the lastname at " "
-                String[] nameParts = vonAndLast.replaceAll("\\s+", " ").trim().split(" ");
-                for (String part : nameParts) {
-                    // use first character of each part of lastname
-                    alphaStyle.append(part, 0, 1);
+            int maxAuthors = Math.min(numberOfAuthors, MAX_ALPHA_AUTHORS);
+            for (int i = 0; i < maxAuthors; i++) {
+                String lastName = getLastName(authorList.getAuthor(i));
+                alphaStyle.append(lastName, 0, 1);
+                if (alphaStyle.length() >= 4) {
+                    // Stop after 4 authors
+                    break;
                 }
             }
-            if (andOthersPresent || maxAuthorsExceeded) {
-                alphaStyle.append("+");
-            }
         }
         return alphaStyle.toString();
     }
 
+    static String getLastName(Author author) {
+        String[] nameParts = author.getNamePrefixAndFamilyName()
+                                   .replaceAll("\\s+", " ")
+                                   .trim()
+                                   .split(" ");
+        return nameParts[nameParts.length - 1];
+    }
```

Comment:
Why can't you use `org.jabref.model.entry.Author#getFamilyName`?

Reply:
Just tried it. The tests fail.

Reply:
Changed.